### PR TITLE
Ignorable events

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Ignore events which server indicates are explicitly ignorable by the SDK.

## Why?
Future proofing

## Checklist

1. Part of https://github.com/temporalio/sdk-features/issues/143

2. How was this tested:
Added UT

3. Any docs updates needed?

### CODE REVIEW

**Comment by Sushisource** [comment]
For reviewers: Skip the first commit

File: core/src/core_tests/workflow_tasks.rs [hunk 1]
```diff
@@ -2054,3 +2054,36 @@ async fn continue_as_new_preserves_some_values() {
...
         .unwrap();
 }
+
+#[rstest]
+#[tokio::test]
+async fn ignorable_events_are_ok(#[values(true, false)] attribs_unset: bool) {
+    let mut t = TestHistoryBuilder::default();
+    t.add_by_type(EventType::WorkflowExecutionStarted);
+    let id = t.add_get_event_id(
+        EventType::Unspecified,
```
Comment (bergundy) [inline core/src/core_tests/workflow_tasks.rs:2064]: Do event types that aren't in the compiled proto definitions end up as `Unspecified`?

File: core/src/core_tests/workflow_tasks.rs [hunk 2]
```diff
@@ -2054,3 +2054,36 @@ async fn continue_as_new_preserves_some_values() {
...
+            history_event::Attributes::WorkflowPropertiesModifiedExternallyEventAttributes(
+                Default::default(),
+            ),
+        ),
+    );
+    t.modify_event(id, |e| e.worker_may_ignore = true);
+    if attribs_unset {
+        t.modify_event(id, |e| {
+            e.event_type = EventType::WorkflowPropertiesModifiedExternally as i32;
+            e.attributes = None;
```
Comment (bergundy) [inline core/src/core_tests/workflow_tasks.rs:2075]: It makes sense to me for unknown enum variants end up as `None`s.

**Review by bergundy — APPROVED** [review]
LGTM.
Just making sure that we've tested what adding a new enum variant for the attributes and type would actually look like.

**Review by cretz — APPROVED** [review]

File: core/src/core_tests/workflow_tasks.rs (hunk 1 above)
Comment (Sushisource) [reply core/src/core_tests/workflow_tasks.rs:2064]: Yes

**Pull request merged**